packages/ogmios/src/CardanoNode/cardanoNode.ts

```diff
@@ -0,0 +1,49 @@
+import { Cardano, CardanoNodeError, CardanoNodeFailure } from '@cardano-sdk/core';
+import {
+  ConnectionConfig,
+  ServerNotReady,
+  WebSocketClosed,
+  createInteractionContext,
+  createStateQueryClient
+} from '@cardano-ogmios/client';
+import { Logger, dummyLogger } from 'ts-log';
+import { mapEraSummary } from './mappers';
+
+/**
+ * Connects to the cardano node through ogmios
+ *
+ * @param {ConnectionConfig} connectionConfig Ogmios connection configuration
+ * @param {Logger} logger object implementing the Logger abstract class
+ * @returns {Cardano.CardanoNode} CardanoNode instance
+ * @throws {CardanoNodeError}
+ */
+export const cardanoNode = (connectionConfig: ConnectionConfig, logger: Logger = dummyLogger): Cardano.CardanoNode => {
+  const getInteractionContext = async (methodName: string) =>
+    createInteractionContext(
+      (error) => {
+        logger.error.bind(logger)({ error: error.name, module: `cardanoNode.StateQuery.${methodName}` }, error.message);
+      },
+      logger.info.bind(logger),
+      { connection: connectionConfig, interactionType: 'LongRunning' }
+    );
+
+  return {
+    StateQuery: {
+      eraSummaries: async () => {
+        const interactionContext = await getInteractionContext('eraSummaries');
```

Comment:
Better to share a single interaction context across all queries, so best to convert this to a class and follow https://github.com/input-output-hk/cardano-graphql/blob/master/packages/api-cardano-db-hasura/src/CardanoNodeClient.ts for inspiration. There's no need to introduce the retry logic here, as we're expecting the process manager to only startup the services when Ogmios is ready.

Reply:
Done in d43ad7b75626b7eacdfbf0e21296706e27acea08. I had to change `CardanoNode` interface as it is a class now though